src/Microsoft.Diagnostics.Runtime/src/Windows/Minidump.cs

```diff
@@ -198,7 +198,11 @@ private async Task<ThreadReadResult> ReadThreadData(Stream stream)
                             continue;
 
                         int count = ResizeBytesForArray<MinidumpThread>(numThreads, ref buffer);
+#if NETCOREAPP3_1
```

Comment:
What does converting buffer to Memory<T> buy us here?

I am trying to avoid littering the code with `#if [version]` #defines.  Is there a way to make this into an extension method of some sort?  I'm not sure what that would look like.  (It would fit here: https://github.com/microsoft/clrmd/blob/master/src/Microsoft.Diagnostics.Runtime/src/Extensions/SpanExtensions.cs#L20).

If there's a clear pattern to doing this it'd really need to be spread throughout the codebase, and I'd rather not duplicate this #if pattern everywhere.  (Maybe these are the only places, but I'm _never_ going to remember `AsMemory` and splitting the codepath when reading from streams.)

Reply:
An extension method makes sense. Agree that `#if` is ugly and good to remove.

> What does converting buffer to Memory buy us here?

I will look into whether this is actually a useful change for those frameworks. If not we can disable the diagnostic or suppress these occurrences.

> I'm never going to remember AsMemory and splitting the codepath when reading from streams.

The good news is you don't have to. The analyzer will nag you with a warning. All these changes are a result of analyzer warnings.

If it isn't already, I'd consider making CI builds fail on warnings.

Reply:
The rule's description suggests there is a perf improvement to using the memory overload. From https://docs.microsoft.com/en-gb/visualstudio/code-quality/ca1835?view=vs-2019:

> The memory-based method overloads have a more efficient memory usage than the byte array-based ones.

Adding an extension method addresses the `string.Contains`/`IndexOf` comment above. It's not clear to me how to avoid the `#if` here with an extension method.

So it's your call as to whether you wish to keep the `#if`, suppress the warnings and keep the old code, or disable the rule altogether.

Reply:
Thanks for checking!  Let's leave your change as-is since there's a real perf benefit here.  I'll try to figure out whether I want to refactor the codepaths better in the future.